docs/artifact_repositories/nuget.md

```diff
@@ -0,0 +1,100 @@
+# Phylum NuGet Registry
+
+The Phylum NuGet registry is based on NuGet's [Server API].
+
+[Server API]: https://learn.microsoft.com/en-us/nuget/api/overview
+
+## Configuration
+
+All configuration options will require a Phylum API key, since Phylum requires
+authentication. You can find out how to generate one in our
+[API Keys documentation].
+
+In the following examples, all API keys will be represented as
+`<PHYLUM_API_KEY>`, so make sure to replace them with your generated key.
+
+Additionally, if the default policy is not sufficient, a group can be passed to
+evaluate all packages against the group's policy. To do this, just replace
+`<PHYLUM_ORG>` and `<PHYLUM_GROUP>` with the desired org and group name. The
+supplied API key **must** have access to this group.
+
+If Phylum's default [policy] is sufficient, you can omit the username from the
+authentication details.
+
+[API Keys documentation]: ../knowledge_base/api-keys.md#generate-an-api-key
+[policy]: ../knowledge_base/policy.md
+
+### `dotnet`
+
+To use the Phylum NuGet registry, the original nuget.org registry first needs to
+be disabled:
+
+```sh
+dotnet nuget disable source nuget.org
+```
+
+Once the official registry is disabled, the Phylum source can be added. If
+you're not running Windows, you'll also have to add the
+`--store-password-in-clear-text` flag.
+
+```sh
+dotnet nuget add source http://localhost:8080/v3/index.json \
+    --protocol-version 3 \
+    --name Phylum \
+    --valid-authentication-types basic \
+    --username <PHYLUM_ORG>/<PHYLUM_GROUP> \
+    --password <PHYLUM_API_KEY>
+```
+
+> ⚠️ **WARNING** ⚠️
+>
+> Do not accidentally save your token into your shell history.
```

Comment:
This seems confusing and is probably unnecessary. If you run the above command, the token will be briefly visible in command line arguments (or persisted if something is auditing process starts), persisted in the shell history unless you've done something to avoid that (but we don't suggest any solution), and persisted in your NuGet config file.

If you use `--store-password-in-clear-text`, the token will be persisted in plain text somewhere near your shell history. If not, the password will most likely be stored in a way that at least any process running as your user account can decrypt without user interaction. I checked and unfortunately [NuGet still does not support storing the password in a keyring](https://github.com/dotnet/dotnet/blob/ff9c1c1768071e85fb600884bdf764d7779d15a9/src/nuget-client/src/NuGet.Core/NuGet.Configuration/Utility/EncryptionUtility.cs#L15-L38), so if we have any non-Windows users they will be persisting their token in plain text, and I think that's already the case for the other supported registry APIs as well.

Reply:
Persisting the token in plain text is fine if that's what the ecosystem decides is appropriate. This is a standard warning which makes sense and is easily actionable for the user.